modules/transformer-kinesis/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/transformer/kinesis/parquet/Codecs.scala

```diff
@@ -54,14 +58,50 @@ private [parquet] object Codecs {
       case FieldValue.StructValue(values) =>
         values
           .foldLeft[RowParquetRecord](RowParquetRecord.empty) {
-            case (acc, value) =>
-              acc.add(value._1, value._2, config)
+            case (acc, NamedValue(name, value)) =>
+              acc.add(name, value, config)
           }
     }
 
+    private def encodeDecimal(value: BigDecimal, precision: DecimalPrecision) = {
+      precision match {
+        case DecimalPrecision.Digits9 =>
+          as[Int](value.underlying().unscaledValue().intValue())
+        case DecimalPrecision.Digits18 =>
+          as[Long](value.underlying().unscaledValue().longValueExact())
+        case DecimalPrecision.Digits38 =>
+          encodeDecimalAsByteArray(value)
+      }
+    }
+
+    /**
+     * Inspired by org.apache.spark.sql.execution.datasources.parquet.ParquetWriteSupport.makeDecimalWriter
+     * and com.github.mjakubowski84.parquet4s.Decimals.binaryFromDecimal
+     */
+    private def encodeDecimalAsByteArray(value: BigDecimal) = {
+      val unscaledBytes = value.underlying().unscaledValue().toByteArray
+      val bytesDifference = ParquetSchema.byteArrayLength - unscaledBytes.length
+      if (bytesDifference == 0) {
+        BinaryValue(unscaledBytes)
+      } else {
+        val buffer = ByteBuffer.allocate(ParquetSchema.byteArrayLength)
+        val sign: Byte = if (unscaledBytes.head < 0) -1 else 0
```

Comment:
Please confirm there is no chance the `unscaledBytes` might have length of zero?

Reply:
I was worried about the zero value, but it looks like you're correct:

```
scala> val bd = BigDecimal("0")
bd: scala.math.BigDecimal = 0

scala> bd.underlying.unscaledValue.toByteArray
res0: Array[Byte] = Array(0)
```